state-reconstruct-fetcher/src/blob_http_client.rs

```diff
@@ -0,0 +1,107 @@
+use serde_json::Value;
+use tokio::time::{sleep, Duration};
+
+use crate::types::ParseError;
+
+/// `MAX_RETRIES` is the maximum number of retries on failed blob retrieval.
+const MAX_RETRIES: u8 = 5;
+/// The interval in seconds to wait before retrying to fetch a blob.
+const FAILED_FETCH_RETRY_INTERVAL_S: u64 = 10;
+
+pub struct BlobHttpClient {
+    client: reqwest::Client,
+    url_base: String,
+}
+
+impl BlobHttpClient {
+    pub fn new(blob_url: String) -> eyre::Result<Self> {
+        let mut headers = reqwest::header::HeaderMap::new();
+        headers.insert(
+            "Accept",
+            reqwest::header::HeaderValue::from_static("application/json"),
+        );
+        let client = reqwest::Client::builder()
+            .default_headers(headers)
+            .build()?;
+        Ok(Self {
+            client,
+            url_base: blob_url,
+        })
+    }
+
+    pub async fn get_blob(&self, kzg_commitment: &[u8]) -> Result<Vec<u8>, ParseError> {
```

Comment:
We can use the `let else` pattern here too.

Reply:
How? All the matches in `get_blob` use both Ok and Err data...

Reply:
I'll send in a suggestion instead!